matrix/splunk-otel-dotnet-metadata.yaml

```diff
@@ -0,0 +1,564 @@
+    default: 'info'
+    type: string
+    category: diagnostic logging
+  - env: OTEL_DOTNET_AUTO_LOG_DIRECTORY
+    description: Directory of the .NET tracer logs. The default value is `/var/log/opentelemetry/dotnet` for Linux, and `%ProgramData%\OpenTelemetry .NET AutoInstrumentation\logs` for Windows. Can't be set using the web.config or app.config files.
+    default: ''
+    type: string
+    category: diagnostic logging
+  - env: OTEL_DOTNET_AUTO_TRACES_CONSOLE_EXPORTER_ENABLED
+    description: Whether the traces console exporter is activated. The default value is `false`.
+    default: 'false'
+    type: boolean
+    category: diagnostic logging
+  - env: OTEL_DOTNET_AUTO_METRICS_CONSOLE_EXPORTER_ENABLED
+    description: Whether the metrics console exporter is activated. The default value is `false`.
+    default: 'false'
+    type: boolean
+    category: diagnostic logging
+  - env: OTEL_DOTNET_AUTO_LOGS_CONSOLE_EXPORTER_ENABLED
+    description: Whether the logs console exporter is activated. The default value is `false`.The default value is `false`.
+    default: 'false'
+    type: boolean
+    category: diagnostic logging
+  - env: OTEL_DOTNET_AUTO_LOGS_INCLUDE_FORMATTED_MESSAGE
+    description: Whether the log state have to be formatted. The default value is `false`.
+    default: 'false'
+    type: boolean
+    category: diagnostic logging
+instrumentations:
+  - keys:
+      - ASPNET
+    instrumented_components:
+      - name: ASP.NET Framework (.NET Framework)
+        supported_versions: See general requirements
+    description: MVC / WebApi (Only integrated pipeline mode supported). Metrics requires trace instrumentation.
+    stability: beta
+    support: community
+    dependencies:
+      - name: ASP.NET Instrumentation for OpenTelemetry
+        source_href: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.AspNet
+        package_href: https://www.nuget.org/packages/OpenTelemetry.Instrumentation.AspNet
+        version: 1.0.0-rc9.9
+        stability: beta
+      - name: ASP.NET Telemetry HttpModule for OpenTelemetry
+        source_href: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule
+        package_href: https://www.nuget.org/packages/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule
+        version: 1.0.0-rc9.9
+        stability: beta
+    signals:
+      - traces: 
+      - metrics:
+          - id: http.server.duration_{bucket|count|sum}
+            instrument: Cumulative counters (histogram)
+            description: Duration of the inbound HTTP request, in the form of count, sum, and histogram buckets. This metric originates multiple metric time series, which might result in increased data ingestion costs.
```

Comment:
I see 2 problems with that: one is that this is already in the upstream opentelemetry semantic conventions, and instead of duplicating them we should simply reference them by the instrument id/name. Two: the `_{bucket|count|sum}` histogram split is not something we should document here, in the instrumentation layer -- the backend is responsible for that, and we should not hardcode this logic here (especially when it will change in the following months).

Reply:
Lets go through first topic:
I think that we neeed to support both, by reference to the sem-conv:

```
      - metrics:
          - id: metric.http.server.request.duration
            sem-conv-href: hredhttps://github.com/open-telemetry/semantic-conventions/blob/v1.22.0/model/metrics/http.yaml
```

but also by local defition of all variables:

```
      - metrics:
          - metric_name: nservicebus.messaging.successes // I am changing attribute id to metric_name ere, as in sem-conv-href: hredhttps://github.com/open-telemetry/semantic-conventions/blob/v1.22.0/model/metrics/http.yaml it is separate
            instrument: Cumulative counter
            description: Number of messages successfully processed by the endpoint.
```

Reply:
@theletterf, what do you think? Are you ok with keeping it here like `http.server.duration` instead of `http.server.duration_{bucket|count|sum}`

Reply:
> Lets go through first topic: I think that we neeed to support both, by reference to the sem-conv:
> 
> ```
>       - metrics:
>           - id: metric.http.server.request.duration
>             sem-conv-href: hredhttps://github.com/open-telemetry/semantic-conventions/blob/v1.22.0/model/metrics/http.yaml
> ```

LGTM 👍

Reply:
I think we can just use the description to describe the full behavior of the setting. That, or produce a different setting entry for each case (`http_sum`, `http_bucket`, etc.)

I'm not sure about the `sem-conv-href` field. Docs team wouldn't probably use it, and it seems entirely manually? (Also, there's an `hred` in the value that seems extraneous.)

Reply:
> I'm not sure about the `sem-conv-href` field. Docs team wouldn't probably use it, and it seems entirely manually? (Also, there's an `hred` in the value that seems extraneous.)

The point behind this is to avoid duplicating the OTel spec when we simply implement it, and not deviate from it. The docs team should _absolutely_ use these fields, to download the actual definition of a metrics. This is just a link to another yaml file basically.

Reply:
> I think we can just use the description to describe the full behavior of the setting. That, or produce a different setting entry for each case (`http_sum`, `http_bucket`, etc.)
> 

I think that it can be done on documentation side. As I understand, all `histogram`s behaves in the same way.